pkg/fetch/endoflife-date/products/types.go

```diff
@@ -0,0 +1,107 @@
+		Property    string  `yaml:"property" json:"property"`
+		Position    string  `yaml:"position" json:"position"` // enum: "after-release-column", "before-latest-column", "after-latest-column"
+		Label       string  `yaml:"label" json:"label"`
+		Description *string `yaml:"description,omitempty" json:"description,omitempty"`
+		Link        *string `yaml:"link,omitempty" json:"link,omitempty"`
+	} `yaml:"customColumns,omitempty" json:"custom_columns,omitempty"`
+	Identifiers []struct {
+		Repology *string `yaml:"repology,omitempty" json:"repology,omitempty"`
+		CPE      *string `yaml:"cpe,omitempty" json:"cpe,omitempty"`
+		PURL     *string `yaml:"purl,omitempty" json:"purl,omitempty"`
+	} `yaml:"identifiers,omitempty" json:"identifiers,omitempty"`
+	Auto *struct {
+		Cumulative *bool `yaml:"cumulative,omitempty" json:"cumulative,omitempty"`
+		Methods    []any `yaml:"methods" json:"methods"`
+	} `yaml:"auto,omitempty" json:"auto,omitempty"`
+	Releases []struct {
+		ReleaseCycle       string         `yaml:"releaseCycle" json:"release_cycle"`
+		Link               *string        `yaml:"link,omitempty" json:"link,omitempty"`
+		ReleaseLabel       *string        `yaml:"releaseLabel" json:"release_label,omitempty"`
+		EOLColumn          any            `yaml:"eolColumn" json:"eol_column,omitempty"`                   // string or bool
+		EOL                any            `yaml:"eol" json:"eol,omitempty"`                                // string<date> or bool
+		EOASColumn         any            `yaml:"eoasColumn" json:"eoas_column,omitempty"`                 // string or bool
+		EOAS               any            `yaml:"eoas" json:"eoas,omitempty"`                              // string<date> or bool
+		EOESColumn         any            `yaml:"eoesColumn" json:"eoes_column,omitempty"`                 // string or bool
+		EOES               any            `yaml:"eoes" json:"eoes,omitempty"`                              // string<date> or bool
+		DiscontinuedColumn any            `yaml:"discontinuedColumn" json:"discontinued_column,omitempty"` // string or bool
+		Discontinued       any            `yaml:"discontinued" json:"discontinued,omitempty"`              // string<date> or bool
+		ReleaseDateColumn  any            `yaml:"releaseDateColumn" json:"release_date_column,omitempty"`  // string or bool
+		ReleaseDate        *string        `yaml:"releaseDate" json:"release_date,omitempty"`               // string<date>
+		ReleaseColumn      any            `yaml:"releaseColumn" json:"release_column,omitempty"`           // string or bool
+		Latest             *string        `yaml:"latest" json:"latest,omitempty"`
+		LatestReleaseDate  *string        `yaml:"latestReleaseDate" json:"latest_release_date,omitempty"` // string<date>
+		Extras             map[string]any `yaml:",inline,omitempty" json:"extras,omitempty"`
+	} `yaml:"releases" json:"releases"`
+	Extras map[string]any `yaml:",inline,omitempty" json:"extras,omitempty"`
+}
+
+type AutoMethodGit struct {
```

Comment:
The PR diff looks nice!
Just for curiosity, these AutoMethod* structs are not used currently, but it seems nice if used.  How will they be used? At extract time?

Reply:
I plan to use this at the time of `vuls-data-update extract`.
First, I'm just transcribing the type according to the json schema.
https://github.com/endoflife-date/endoflife.date/blob/2628d65d86b8472c879dd692b5047417f722bce2/product-schema.json#L87

Reply:
Sounds nice!